### ISSUE TITLE
homebridge-zp crashes Homebridge

### ISSUE THREAD

**Issue by cakefornoreason** [ebaauw/homebridge-zp#27]
Ever since installing these last couple of updates, I'm unable to get Homebridge to work with homebridge-zp configured. I'm on the recommended node version, v8.9.4.

Here's the error message I get, which appears to pertain to alarms:

TypeError: alarmClock.CurrentAlarmList is not iterable
    at ZPPlatform.<anonymous> (/usr/local/lib/node_modules/homebridge-zp/lib/ZPPlatform.js:365:54)
    at self._parseKey (/usr/local/lib/node_modules/homebridge-zp/node_modules/sonos/lib/services/AlarmClock.js:81:14)
    at parseString (/usr/local/lib/node_modules/homebridge-zp/node_modules/sonos/lib/services/Service.js:108:12)
    at Parser.<anonymous> (/usr/local/lib/node_modules/homebridge-zp/node_modules/xml2js/lib/parser.js:303:18)
    at emitOne (events.js:116:13)
    at Parser.emit (events.js:211:7)
    at SAXParser.onclosetag (/usr/local/lib/node_modules/homebridge-zp/node_modules/xml2js/lib/parser.js:261:26)
    at emit (/usr/local/lib/node_modules/homebridge-zp/node_modules/sax/lib/sax.js:624:35)
    at emitNode (/usr/local/lib/node_modules/homebridge-zp/node_modules/sax/lib/sax.js:629:5)
    at closeTag (/usr/local/lib/node_modules/homebridge-zp/node_modules/sax/lib/sax.js:889:7)

Is this a code bug or possibly something in my setup causing the issue? Thanks in advance!

**Comment by ebaauw** [comment]
They changed the AlarmClock interface in sonos, the package I use to interact with the Sonos players.  Maybe I missed one.  What version of homebridge-zp are you on?  Could you check the version of sonos as well through `npm -g list --depth=1`?

**Comment by cakefornoreason** [comment]
I'm using the newest (just a few hours old) version of homebridge-zp, and the sonos version listed in my setup is 0.22.1

**Comment by ebaauw** [comment]
Ah, I can reproduce the error.  You don't have any alarms defined on your Sonos, do you?  As a workaround you could just define a dummy one (just leave it disabled), and the error will go away.  I'll check what's causing this and come up with a fix.

Unfortunately, the current code is so complex, that I need to make the call to get the alarms, even when `alarms` is not set in config.json.

**Comment by cakefornoreason** [comment]
Hmm... the fix of adding an alarm and disabling it doesn't seem to fix it for me. I'll keep checking, but alternately, I've been using a config.json that just disables the platform and that works fine, of course.

If there's anything else I can try out for you, I'd be glad to help out. Thanks!

**Comment by cakefornoreason** [comment]
Sorry, didn't mean to call it a "fix." What I mean is that I still get the crash even with an alarm present, so at least for me, adding an alarm doesn't appear to work as a workaround, but again, it could be some other part of my specific config. Fortunately, the Sonos/HomeKit integration isn't critical for me. Good catch on finding the source though! I'll stay tuned :)

**Comment by ebaauw** [comment]
v0.2.16 will only crash when `"alarms"` is set in config.json and there's less than two alarms on Sonos.  I've issued a PR to node-sonos (https://github.com/bencevans/node-sonos/pull/212) to fix that, but we'll have wait and see if they accept it.

**Comment by ebaauw** [comment]
> Fortunately, the Sonos/HomeKit integration isn't critical for me.

Yeah, less critical to me as well.  Because of that, I've spent far less effort on homebridge-zp to make it bullet proof like homebridge-hue.  Every advantage has its disadvantage...

**Comment by ebaauw** [comment]
OK, the fix is included in sonos@0.22.2.  If you re-install homebridge-zp, npm should update sonos:
```
$ npm -g list --depth=1
/usr/local/lib
├─┬ homebridge@0.4.33
│ ├── chalk@1.1.3
│ ├── commander@2.8.1
│ ├── hap-nodejs@0.4.36
│ ├── node-persist@0.0.8
│ ├── qrcode-terminal@0.11.0
│ └── semver@5.0.3
├─┬ homebridge-zp@0.2.16
│ ├── request@2.83.0
│ ├── semver@5.4.1
│ ├── sonos@0.22.1
│ └── xml2js@0.4.19
...
$ sudo npm -g i homebridge-zp
Password:
+ homebridge-zp@0.2.16
updated 2 packages in 2.753s
$ npm -g list --depth=1
/usr/local/lib
├─┬ homebridge@0.4.33
│ ├── chalk@1.1.3
│ ├── commander@2.8.1
│ ├── hap-nodejs@0.4.36
│ ├── node-persist@0.0.8
│ ├── qrcode-terminal@0.11.0
│ └── semver@5.0.3
├─┬ homebridge-zp@0.2.16
│ ├── request@2.83.0
│ ├── semver@5.4.1
│ ├── sonos@0.22.2
│ └── xml2js@0.4.19
...
```

**Comment by cakefornoreason** [comment]
Updated version works like a charm. Thanks for looking into this!
